tests/functional/test_snap_dcgm.py

```diff
@@ -0,0 +1,17 @@
+import subprocess
+from time import sleep
+
+
+def test_dcgm_exporter_endpoint():
+    """Smoke test of the dcgm-exporter service and its endpoint."""
+    dcgm_exporter_endpoint = "localhost:9400/metrics"
+    subprocess.run(["sudo", "snap", "disconnect", "dcgm:network-bind"])
```

Comment:
non-blocking: this should be ok, but not what I was trying to convey: my suggestion was to more simply design a test that would catch the type of failure we had seen when we had forgotten to include
```
    plugs:                                                                              
      - network-bind   
```

So something like:
1. write a test
2. delete `network-bind` from snapcraft.yaml
3. run functest, ensure your test fails
4. re-add `network-bind` to snapcraft.yaml
5. commit

Reply:
Do you expect the behaviour to be different in those two cases? I can replicate the same pattern by first removing the `network-bind` and then starting the service. Writing a test that removes something from the config feels a little strange. But I guess the test itself is a little weird. Tell me what you think, thanks

Reply:
IMO this approach of disconnecting the interface is easier and cleaner than building the snap without the interface.